test/posix_time/testmicrosec_time_clock.cpp

```diff
@@ -26,39 +52,32 @@ main()
   int i = 0;
   for (i = 0; i<max; i++)
   {
-    for (int j=0; j<100000; j++)
-    {
-      // some systems loop too fast so "last is less" tests fail
-      // due to 'last' & 't2' being equal. These calls slow
-      // it down enough to make 'last' & 't2' different
-#if defined(BOOST_HAS_GETTIMEOFDAY)
-      timeval tv;
-      gettimeofday(&tv, 0);
-#endif
-#if defined(BOOST_HAS_FTIME)
-      SYSTEMTIME st;
-      GetSystemTime(&st);
-#endif
-    }
+    // Some systems loop too fast so "last is less" tests fail due to
+    // 'last' & 't2' being equal. These calls slow it down enough to
+    // make 'last' & 't2' different. Moreover, we must wait the next
+    // second to avoid a change in hour, minute or second field
+    // between acquisition of t1 and t2.
+    sync_to_next_second();
 
     ptime t1 = second_clock::local_time();
     std::cout << to_simple_string(t1) << std::endl;
 
     ptime t2 = microsec_clock::local_time();
     std::cout << to_simple_string(t2) << std::endl;
+
     check("hours match", t1.time_of_day().hours() == t2.time_of_day().hours());
-    check("minutes match", 
-          t1.time_of_day().minutes() == t2.time_of_day().minutes());
-    check("seconds match", 
+    check("minutes match",
           t1.time_of_day().minutes() == t2.time_of_day().minutes());
+    check("seconds match",
+          t1.time_of_day().seconds() == t2.time_of_day().seconds());
     check("hours date", t1.date() == t2.date());
```

Comment:
Not your bug, but description is a big funky here -- really should be something like "check date is the same"

Reply:
I can add a commit to the pull request to change check strings if you want?

Reply:
sure might as well 👍